src/nvim/eval.c

```diff
@@ -20966,6 +21068,8 @@ void ex_function(exarg_T *eap)
 
   /*
    * ":function func" with only function name: list function.
+   * If bang is given, include bang in function head and exclude line numbers
+   * from function body.
```

Comment:
should have an explicit test for this

Reply:
The bang is mainly for the restore functionality of the context, in case some function with the same name exists. Shall there be some flag in `nvim_load_context` and `ctxpop` to explicitly tell whether to override or not? I think this is the point of the context so no need for that maybe

Reply:
True. It's a little weird, but I don't have a better suggestion. Let's leave it.